utils/io/content/contentreader.go

```diff
@@ -0,0 +1,158 @@
+package content
+
+import (
+	"bufio"
+	"encoding/json"
+	"errors"
+	"io"
+	"log"
+	"os"
+	"sync"
+
+	"github.com/jfrog/jfrog-client-go/utils"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+)
+
+type ContentReader struct {
```

Comment:
Let's add a description for this cimponent. Describe the problem it solves and how it works in general.